src/views/pages/tools/purge.html

```diff
@@ -1,23 +1,186 @@
+					</div>
+					<div class="col-md-2">
+						<a on-click="@this.purge()" class-loading="purging" class="btn btn-featured">
+							{{#if purging}}
+								<div class="load-animation"></div>
+							{{else}}
+								PURGE NOW
+							{{/if}}
+						</a>
+						{{#if requestFailed}}
+							<label class="error">An error occurred while purging!</label>
+						{{/if}}
+					</div>
+				</div>
+				<div id="recaptcha-service" class="g-recaptcha" data-callback="recaptchaCallback" data-sitekey="6LdXrfkUAAAAAFkzRiSAV4MBvUfNR1aynq1ggATK"></div>
+				<script type="text/javascript" src="https://www.google.com/recaptcha/api.js?hl=en"></script>
+			</section>
+			<section class="results">
+				{{#if results.length}}
+					<table>
+						<tr>
+							<th>CDN Name</th>
+							<th>Message</th>
+						</tr>
+						{{#each results}}
+							<tr>
+								<td>{{this.name}}</td>
+								<td>
+									{{#if this.result}}
+										<i class="fa fa-check success"></i>
+										<span>Success</span>
+									{{else}}
+										<i class="fa fa-times failed"></i>
+										<span>Failed</span>
+									{{/if}}
+								</td>
+							</tr>
+						{{/each}}
+					</table>
+				{{/if}}
+			</section>
+		</div>
 	</div>
+
 </r-page>
 
 <script>
 	component.exports = {
 		data () {
 			return {
 				title: 'Purge CDN cache - jsDelivr',
+				purging: false,
+				requestFailed: false,
+				recaptchaResponse: '',
+				results: [],
+				purgeChecks: [
+					{
+						provider: 'fastly',
+						check: (result) => {
+							try {
+								result.forEach((res) => {
+									if (res.status !== 'ok') {
+										return false;
+									}
+								});
+							} catch (error) {
+								return false;
+							}
+
+							return true;
+						},
+					},
+					{
+						provider: 'maxcdn',
+						check: (result) => {
+							return result.code === 200;
+						},
+					},
+					{
+						provider: 'cloudflare',
+						check: (result) => {
+							if (typeof result !== 'boolean') {
+								return false;
+							}
+
+							return result;
+						},
+					},
+					{
+						provider: 'quantil',
+						check: (result) => {
+							try {
+								return result.includes('success');
+							} catch (error) {
+								return false;
+							}
+						},
+					},
+				],
 			};
 		},
+		oninit () {
+			if (!Ractive.isServer) {
+				window.recaptchaCallback = undefined;
+
+				$(document).ready(() => {
+					window.recaptchaCallback = function recaptchaCallback (response) {
+						this.set('recaptchaResponse', response);
+					};
+				});
+			}
+		},
+		purge () {
+			this.set('purging', true);
+			this.set('requestFailed', false);
+
+			let paths = this.get('urls').replace(/\s+/g, '');
+
+			if (paths === '') {
+				this.set('purging', false);
+				return;
+			}
+
+			paths = paths.split(',');
+
+			// If there are more than 10 paths, use the captcha endpoint
+			let apiEndpoint = 'https://purge.jsdelivr.net';
+			let postBody = { path: paths };
+
+			if (paths.length > 10) {
```

Comment:
This should be used always.

Reply:
Do you mean always use the CAPTCHA endpoint?

Reply:
Yes.